File: seed/php-sdk/oauth-client-credentials/src/Auth/Types/TokenResponse.php
<?php

namespace Seed\Auth\Types;

use Seed\Core\SerializableType;
use Seed\Core\JsonProperty;

/**
* An OAuth token response.
 */
class TokenResponse extends SerializableType
{
    #[JsonProperty("access_token")]
    /**
     * @var string $accessToken
     */
    public string $accessToken;

    #[JsonProperty("expires_in")]
    /**
     * @var int $expiresIn
     */
    public int $expiresIn;

    #[JsonProperty("refresh_token")]
    /**
     * @var ?string $refreshToken
     */
    public ?string $refreshToken;

    /**
     * @param string $accessToken
     * @param int $expiresIn
     * @param ?string $refreshToken
     */
    public function __construct(
        string $accessToken,
        int $expiresIn,
        ?string $refreshToken = null,
    ) {
        $this->accessToken = $accessToken;
        $this->expiresIn = $expiresIn;
        $this->refreshToken = $refreshToken;
    }
}

File: seed/ruby-sdk-v2/oauth-client-credentials-with-variables/lib/token_response.rb
# frozen_string_literal: true

module Auth
    module Types
        # An OAuth token response.
        class TokenResponse < Internal::Types::Model
            field :access_token, String, optional: true, nullable: true
            field :expires_in, Integer, optional: true, nullable: true
            field :refresh_token, Array, optional: true, nullable: true
        end
    end
end
